[FIX] account: include inactive tags in ```_get_tag_mapper```

apan-odoo opened the pull request:
After this commit: [commit](https://github.com/odoo/odoo/pull/148866/files)

- If tags belonging to localization are present in the database but with active='f', the user error will still trigger. As mentioned in the commit if tags are deleted by the end user, then updating the localization module will resolve this and will create the new tags for that localization, but in our case if the tag are present in the db but are active='f', so updating the localization module will not help as noupdate will become true, if the enduser makes any changes to it. so updating the localization will not change active='f' and the user error will keep on coming unless the end user manually set it active='t'.

**Steps to reproduce**:
1. Create a database in saas-16.2
2. Install the module account_accountant and a localization, for example l10n_ee.
3. With debug mode on, navigate to the account tags menu
     (Configuration > Accounting > Account Tags).
4. Archive any tag using the archive action.
5. On the accounting settings page, under Fiscal Localization, trigger a user error when clicking the reload button.

This user error is blocking.
By making the record inactive, the user has made the account tag no-update, so upgrading the module does not solve this.

- So, as this mentioned commit is introduced in 16.2, any db migrating to v16.2 or further, when the ```_get_tag_mapper``` is triggered this user error is blocking the migration.

**Solution**:
- This commit adds a context to include the inactive tags here: [here](https://github.com/odoo/odoo/blob/saas-16.2/addons/account/models/chart_template.py#L924-L928)

OPW-3716164
UPG-1297494
[TBG-1074](https://upgrade.odoo.com/web#id=1074&cids=1&menu_id=107&action=178&model=upgrade.request.traceback.group&view_type=form)

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/153944).

apan-odoo commented:
Hello @dbkosky @clbr-odoo,
Can you please have a look at this?
Thank you!

clbr-odoo reviewed (changes requested):
Hello ! 

The fix LGTM thanks :+1:
Just few things (quickly discussed with DAKO so this is our combined review :slightly_smiling_face: ):
1. Please remove the traceback. It's not helpful in the context of a commit message. If you feel it's important to the context, you can leave it in the PR message.
2. Preferably you should have a "steps to reproduce", like in the commit that you linked. Something like:
```
Steps to reproduce:
1. Create a database in saas-16.2
2. Install the module account_accountant and a localization, for example l10n_ee.
3. With debug mode on, navigate to the account tags menu
     (Configuration > Accounting > Account Tags).
4. Archive any tag using the archive action.
5. On the accounting settings page, under Fiscal Localization, trigger a user error when clicking the reload button.

This user error is blocking.
By making the record inactive, the user has made the account tag no-update, so upgrading the module does not solve this.
```

3. Could you please to add a test ? You can do something similar to what we did in `TestChartTemplate.test_signed_and_unsigned_tags_tax` (`create an inactive tag, add it to a tax, call `_deref_account_tags`). This way if we make change to the code we make sure this error cannot happen again.

apan-odoo commented:
> Hello !
> 
> The fix LGTM thanks 👍 Just few things (quickly discussed with DAKO so this is our combined review 🙂 ):
> 
> 1. Please remove the traceback. It's not helpful in the context of a commit message. If you feel it's important to the context, you can leave it in the PR message.
> 2. Preferably you should have a "steps to reproduce", like in the commit that you linked. Something like:
> 
> ```
> Steps to reproduce:
> 1. Create a database in saas-16.2
> 2. Install the module account_accountant and a localization, for example l10n_ee.
> 3. With debug mode on, navigate to the account tags menu
>      (Configuration > Accounting > Account Tags).
> 4. Archive any tag using the archive action.
> 5. On the accounting settings page, under Fiscal Localization, trigger a user error when clicking the reload button.
> 
> This user error is blocking.
> By making the record inactive, the user has made the account tag no-update, so upgrading the module does not solve this.
> ```
> 
> 3. Could you please add a test ? You can do something similar to what we did in `TestChartTemplate.test_signed_and_unsigned_tags_tax` (`create an inactive tag, add it to a tax, call `_deref_account_tags`). This way if we make change to the code we make sure this error cannot happen again.

Hello @clbr-odoo @daka-odoo,
Thank you for the review, updated the commit message as suggested and added the test too. can you please review it?

daka-odoo commented:
Hello,

I think you might have meant to tag dako and not me :)

On Mon, Feb 19, 2024 at 10:02 AM Akash Pandey ***@***.***>
wrote:

> Hello !
>
> The fix LGTM thanks 👍 Just few things (quickly discussed with DAKO so
> this is our combined review 🙂 ):
>
>    1. Please remove the traceback. It's not helpful in the context of a
>    commit message. If you feel it's important to the context, you can leave it
>    in the PR message.
>    2. Preferably you should have a "steps to reproduce", like in the
>    commit that you linked. Something like:
>
> Steps to reproduce:
> 1. Create a database in saas-16.2
> 2. Install the module account_accountant and a localization, for example l10n_ee.
> 3. With debug mode on, navigate to the account tags menu
>      (Configuration > Accounting > Account Tags).
> 4. Archive any tag using the archive action.
> 5. On the accounting settings page, under Fiscal Localization, trigger a user error when clicking the reload button.
>
> This user error is blocking.
> By making the record inactive, the user has made the account tag no-update, so upgrading the module does not solve this.
>
>
>    1. Could you please add a test ? You can do something similar to what
>    we did in TestChartTemplate.test_signed_and_unsigned_tags_tax (create
>    an inactive tag, add it to a tax, call _deref_account_tags`). This way
>    if we make change to the code we make sure this error cannot happen again.
>
> Hello @clbr-odoo <https://github.com/clbr-odoo> @daka-odoo
> <https://github.com/daka-odoo>,
> Thank you for the review, updated the commit message as suggested and
> added the test too. can you please review it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/odoo/odoo/pull/153944#issuecomment-1951988799>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVHGPVDTPYNODHOX64JM5MDYUMIL5AVCNFSM6AAAAABDG6SFNSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNJRHE4DQNZZHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

apan-odoo commented:
> Hello, almost good: you need to check the value of the tag in an self.assertSomething, here you're prooving it's not crashing, not that it's working as intended 😉

Suggested changes are done, could you please review it again. Thank you :raised_hands:

clbr-odoo reviewed (approved):
Good job, thanks !

apan-odoo commented:
Hello @dbkosky @clbr-odoo,
As you have approved this, I think it is ready for r+
We have db which are getting blocked by this.
Thank you for your time :+1:

clbr-odoo commented:
Hello @apan-odoo !
Sorry we should have explained you, we are pre-reviewers, once we're done we send the PR to someone that has r+ rights.
The PR is in his pipe now, he will process it shortly for sure :slightly_smiling_face:

FlorianGilbert reviewed (approved):
@robodoo r+
